dependency/engine.go

```diff
@@ -628,8 +629,16 @@ func (engine *Engine) gotStopped(name string, err error, resourceLog []resourceA
 			// The task should never run again, and can be removed completely.
 			engine.uninstall(name)
 		default:
+			logFn := engine.config.Logger.Errorf
+			if strings.Contains(err.Error(), "try again") {
```

Comment:
is string matching the only way for us to find this? On the controller side we're certainly sending a typed error, is it not getting a proper error code that we can check against?

Reply:
My initial approach was to look for a typed error. I _suspect_ [this](https://github.com/juju/juju/blob/develop/apiserver/common/errors.go#L99) is probably the one we are interested in but since this is a vendored dep I didn't want to import a package that may cause a circular dep.

Reply:
I also discovered [this](https://github.com/juju/juju/blob/develop/apiserver/params/apierror.go#L216-L218) check for try-again codes but that lives in the juju repo and the worker seems to be using the [juju/errors](https://github.com/juju/errors) pkg.